manifests/init.pp

```diff
@@ -142,13 +142,22 @@
       $master_or_package_version = $package_version
     }
 
-    if $master_or_package_version !~ /^\d+\.\d+\.\d+([.-]?\d*|\.\d+\.g[0-9a-f]+)$/ {
-      fail("invalid version ${master_or_package_version} requested")
+    if $::osfamily == 'redhat' {
+      if $master_or_package_version !~ /^\d+\.\d+\.\d+(.+)?$/ {
```

Comment:
you can replace `(.+)?` with `.*` to match everything to the end of the line, since we don't need capture groups here